src/catalog/src/durable.rs

```diff
@@ -231,6 +231,9 @@ pub trait ReadOnlyDurableCatalogState: Debug + Send {
     /// Get a snapshot of the catalog.
     async fn snapshot(&mut self) -> Result<Snapshot, CatalogError>;
 
+    /// Get a snapshot of the collections managed by the storage controller.
+    async fn storage_snapshot(&mut self) -> Result<Snapshot, CatalogError>;
+
```

Comment:
I don't think we should use `Snapshot` as a return type here, it contains maps for every collection. Will the non-storage collections be empty?